apps/calendar/js/views/event_base.js

```diff
@@ -316,7 +268,7 @@ EventBase.prototype = {
     if (id) {
       classList.add(this.UPDATE);
 
-      this._loadModel(id, completeDispatch);
+      this._loadModel(id).then(completeDispatch);
```

Comment:
I think you could clean up the flow control in this `dispatch` function with `yield` :)